app/src/main/java/com/panoprogramowanie/testcoroutines/presentation/MainActivity.kt

```diff
@@ -6,9 +6,20 @@ import android.support.v7.app.AppCompatActivity
 import com.panoprogramowanie.testcoroutines.DependencyInjector
 import com.panoprogramowanie.testcoroutines.R
 import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.coroutines.experimental.CoroutineScope
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.launch
+import kotlin.coroutines.experimental.CoroutineContext
 
-class MainActivity : AppCompatActivity(), MainView {
+class MainActivity : AppCompatActivity(), MainView, CoroutineScope {
 
+    lateinit var job: Job
```

Comment:
Are all our presenters in Kotlin? To use this approach appropriately, we'll need to implement 
```
lateinit var job: Job
override val coroutineContext: CoroutineContext
        get() = baseCoroutineScope + job
```
in a presenter/ViewModel

Reply:
We try to write all new code in Kotlin :)
Our current approach is to have Presenters extend CoroutineScope as you suggested.

Unfortunately, to keep our Presenters testable we need to adjust it a bit...
This pull request suggests moving the scope to Activities/Fragments, and expose suspend functions in presenters. This way scope would be closer to lifecycle